src/models/Group.js

```diff
@@ -0,0 +1,31 @@
+const mongoose = require("mongoose");
+
+const groupSchema = new mongoose.Schema(
+  {
+    groupName: {
+      type: String,
+      required: [true, "please provide a group name!"],
+      unique: true,
+    },
+    owner: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "User",
+    },
+    description: {
+      type: String,
+    },
+    members: [
+      {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+      },
+    ],
+    groupEvents: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "Event",
+    },
```

Comment:
Now that I'm reviewing @msrezaie's [PR](https://github.com/Code-the-Dream-School/ee-prac-team2-back/pull/54), I see that we reference the group on each event. My instinct is that we wouldn't need to store the association between groups and events on both schemas, but I am not accustomed to designing core app data for NoSQL data stores like this, so I'm wondering if this sort of denormalization/duplication of association data is actually advised for Mongo and other document stores. I've asked for guidance from some other mentors, but I'm curious about your thoughts on this!

Reply:
Based on some of our wireframes and use case flows, from a frontend perspective, some details of the events that are part of a group are displayed in the group's page. So my idea for having this `groupEvents` field is just for that. Whenever a user clicks on an event's button, they are taken into an event's page through one of the events endpoints (preferebly /api/v1/events/{eventID}) using its ID coming from `groupEvents`.

Reply:
It seemed cleaner and more easily accessible if we populated a reference to all of a Group's associated events, but I'm interested to see what the best practices are.

Reply:
> some details of the events that are part of a group are displayed in the group's page

@msrezaie we should be able to select events from the events collection based on the group id, though. And we can include the id's or whatever other event data we want in the response for the `GET /groups/:id` endpoint.

I did get a response from another volunteer and it sounds like we should design with [normalization](https://www.databasestar.com/database-normalization/) in mind, just as we would in Postgres or other relational stores. Since the relationship between groups and events looks like a classic ["one-to-many" relationship](https://www.sqltutorial.net/sql-one-to-many-relationship.html) to me (one group may have many events, each event belongs to a group) I think we should just reference the group on the events schema. As mentioned in the link, structuring the relationship this way helps maintain integrity of the data and avoids data redundancy. 

I'll post this in slack too in case we want to discuss more.

Reply:
Hmm interesting, yeah I see and understand your point now. I wasn't really thinking of listing of all groups endpoint in this way honestly. And yes, the `GET /groups/:groupID` should suffice instead of having the `groupEvents` field.

I'll make the proper adjustments for this in my Events PR. And @marice-romero you may omit the `groupEvents` field.